ledger/ledger-utxo/src/main/java/net/corda/v5/ledger/utxo/token/selection/TokenSelection.java

```diff
@@ -21,7 +21,14 @@ public interface TokenSelection {
      * If available, a set of tokens will be selected that sum to at least the target amount specified.
      * The tokens will be locked in the cache to prevent other flows from selecting them.
      *
+     * A unique deterministic identifier for a token claim is provided per API call.
+     * Corda may rerun a token claim request on behalf of the calling flow in the event system instability.
+     * This ID will be used to deduplicate TokenClaims in this scenario.
+     * This ID needs to be unique per TokenClaim within a single flow.
+     *
      * @param criteria The {@link TokenClaimCriteria} used to select tokens.
```

Comment:
I'd wonder about adding an example using a static constant or similar to Ben's https://github.com/corda/corda-api/pull/1574/files#r1538964821 but I wouldn't let it hold up the PR